src/compton/Compton.cc

```diff
@@ -107,25 +123,47 @@ Compton::Compton(const std::string &filehandle,
  * \param etemp The SCALED electron temperature ( temp / electron rest-mass )
  * \return      n_grp x n_grp x n_xi interpolated scattering kernel values
  */
-std::vector<std::vector<std::vector<double>>>
-Compton::interpolate(const double etemp) {
+std::vector<std::vector<std::vector<std::vector<double>>>>
+Compton::interpolate_csk(const double etemp) {
```

Comment:
A 4-D vector seems pretty intense.  We might want to consider a different container since this requires 4 pointer dereferences to access each data element.